hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -226,26 +227,22 @@ public String getRecordKey(T record, Schema schema) {
    * @param metadataMap  A map containing the record metadata.
    * @param schema       The Avro schema of the record.
    * @param orderingFieldName name of the ordering field
-   * @param orderingFieldTypeOpt type of the ordering field
-   * @param orderingFieldDefault default value for ordering
    * @return The ordering value.
    */
   public Comparable getOrderingValue(Option<T> recordOption,
                                      Map<String, Object> metadataMap,
                                      Schema schema,
-                                     String orderingFieldName,
-                                     Option<Schema.Type> orderingFieldTypeOpt,
-                                     Comparable orderingFieldDefault) {
+                                     Option<String> orderingFieldName) {
     if (metadataMap.containsKey(INTERNAL_META_ORDERING_FIELD)) {
       return (Comparable) metadataMap.get(INTERNAL_META_ORDERING_FIELD);
     }
 
-    if (!recordOption.isPresent() || !orderingFieldTypeOpt.isPresent()) {
-      return orderingFieldDefault;
+    if (!recordOption.isPresent() || orderingFieldName.isEmpty()) {
```

Comment:
whats the case where recordOption could be empty? 
are there chances that ordering col is set in options and is of type String (non integer) and recordOption is empty? won't we return 0 (integer) as the default from here.